coderd/httpmw/authz.go

```diff
@@ -35,3 +36,15 @@ func AsAuthzSystem(mws ...func(http.Handler) http.Handler) func(http.Handler) ht
 		})
 	}
 }
+
+// RecordAuthzChecks enables recording all of the authorization checks that
+// occurred in the processing of a request. This is mostly helpful for debugging
+// and understanding what permissions are required for a given action without
+// needing to go hunting for checks in the code, where you're quite likely to
+// miss something subtle or a check happening somewhere you didn't expect.
+func RecordAuthzChecks(next http.Handler) http.Handler {
```

Comment:
Something to include, is that a failed authz check will likely prevent further checks. So it's most helpful when the actions all pass.

It might feel obvious, but worth mentioning imo